client/src/pages/Signup.js

```diff
@@ -1,216 +1,221 @@
-import React, { useState } from "react";
-import { Link, useHistory } from "react-router-dom";
+import React, { useState } from 'react'
+import { Link, useHistory } from 'react-router-dom'
 
 import {
   Button,
   TextField,
   Typography,
   Container,
-  Snackbar,
-} from "@material-ui/core";
+  Snackbar
+} from '@material-ui/core'
 
-import MuiAlert from "@material-ui/lab/Alert";
+import MuiAlert from '@material-ui/lab/Alert'
 
-import { makeStyles } from "@material-ui/core/styles";
+import { makeStyles } from '@material-ui/core/styles'
 
-import axios from "axios";
+import axios from 'axios'
 
-function Alert(props) {
-  return <MuiAlert elevation={6} variant="filled" {...props} />;
+function Alert (props) {
+  return <MuiAlert elevation={6} variant='filled' {...props} />
 }
 
-const useStyles = makeStyles((theme) => ({
+const useStyles = makeStyles(theme => ({
   root: {
     marginTop: theme.spacing(8),
-    display: "flex",
-    flexDirection: "column",
-    alignItems: "center",
-    borderRadius: "10px",
-    boxShadow: "0 0 5px 0",
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    borderRadius: '10px',
+    boxShadow: '0 0 5px 0',
     padding: theme.spacing(3, 5),
-    backgroundColor: "white",
+    backgroundColor: 'white'
   },
   underline: {
-    width: "50px",
-    height: "4px",
-    color: "#32BE72",
-    backgroundColor: "#32BE72",
+    width: '50px',
+    height: '4px',
+    color: '#32BE72',
+    backgroundColor: '#32BE72'
   },
   form: {
     marginTop: theme.spacing(1),
-    width: "100%",
+    width: '100%'
   },
   formButton: {
-    display: "flex",
-    margin: "24px auto 32px",
-    textTransform: "none",
-    backgroundColor: "#32BE72",
-    color: "white",
-    width: "120px",
-    height: "40px",
+    display: 'flex',
+    margin: '24px auto 32px',
+    textTransform: 'none',
+    backgroundColor: '#32BE72',
+    color: 'white',
+    width: '120px',
+    height: '40px'
   },
   link: {
-    textDecoration: "none",
+    textDecoration: 'none',
     color: theme.palette.primary.main
   }
-}));
+}))
 
-const Signup = () => {
-  const classes = useStyles();
-  let history = useHistory();
+const Signup = ({ accountValues }) => {
+  const classes = useStyles()
+  let history = useHistory()
 
   const [values, setValues] = useState({
-    name: "",
-    email: "",
-    password: "",
-    confirmPassword: "",
-  });
+    name: '',
```

Comment:
we need to use a consistent style in managing variables. we shouldn't consider `""` as default in one component, but `undefined` for the same values in another component. we should strive to keep variables a single type when possible, having the same variables represent different types in different components increases the complexity of the program and reduces our ability to reason about it, with no benefits

Reply:
It is a little complicated, but there is a reason I am using undefined in one instance and "" in another. Undefined is meant to be the starting state (where you first enter the application or refresh the page), whereas "" is when the page is rendered and the check to see if any user has logged in has happened; if no one is logged in, the values are set to "".